### ISSUE TITLE
[topological_navigation] Offer service to get map without 'no_go' nodes

### ISSUE THREAD

**Issue by cdondrup** [strands-project/strands_navigation#240]
The mdp planner has a few problems if the first node in the map returned by `/get_topological_map` is a no go node without edges. To have the mdp agnostic of the underlying navigation, would it be possible to have a service that returns only nodes that are not no go nodes?

This actually most likely caused the issue during the aaf test today.

**Comment by cdondrup** [comment]
I guess this is an issue for @Jailander

**Comment by Jailander** [comment]
Hmmm it sounds like a bit of a hack, but yes that should be possible I could also just sort them alphabetically and the first node would be chargingpoint

**Comment by bfalacerda** [comment]
i think i have a solution that doesn't require code changes anywhere: add edges from no go nodes to their neighbouring nodes. would this break anything or your side @Jailander ?

**Comment by Jailander** [comment]
hey, well one of the points of nogo nodes is exactly that they have no edges, I think for now I will exclude the nogo nodes from the service call and sort them, is not a big change and we make sure it doesn't break anything, maybe for the future I can have a topological map/no go map on separate topics policy, but since that would require more changes I'll leave it for after the deployment

**Comment by bfalacerda** [comment]
ok, following @marc-hanheide 's suggestion, i'll filter nodes without edges at the mdp, and nothing needs to be done at the topological navigation level. sounds good @Jailander ?

**Comment by bfalacerda** [comment]
this will have to wait a while though, i'm prioritising g4s now, and you can avoid the issue by either playing with mongo entries, or going to `~/tmp/travel_times_estimator/something_something.mdp` and change the value for the initial waypoint from 0 to an index that corresponding to a non no-go node (if you choose another value randomly, odds are you'll get a good one)

**Comment by Jailander** [comment]
yes sounds good, but since we need the scheduler working at aaf tomorrow I'll do the sorting anyway its harmless and will look nicer

**Comment by marc-hanheide** [comment]
OK, can you tell me once again, what we would have to change on mongo?

**Comment by marc-hanheide** [comment]
ah, ok, @Jailander seems to know

**Comment by bfalacerda** [comment]
if you edit a topological map entry in mongo, then the order you retrieve the topological nodes with change, and the first node should stop being a no go node

**Comment by Jailander** [comment]
wait now I'm confused, aren't you using `/topological_map_publisher/get_topological_map` for that?, my plan was to sort that, anyway, modifying the mongodb works too but I prefer to make sure by modifying the service in case we move a no_go node and we have the same problem again

**Comment by marc-hanheide** [comment]
what are you sorting according to? putting non-no_go nodes first?

**Comment by Jailander** [comment]
well alphabetically, that would make _ChargingPoint_ the first one all no go nodes are Waypoint7*

**Comment by bfalacerda** [comment]
that should work as a hot fix.

**Comment by marc-hanheide** [comment]
OK, so, also a hack, but a slightly more robust one ;-) But go for it. We need this for tomorrow's test

**Comment by Jailander** [comment]
Yes #241 is a hack after the deployment I will go for having the topological map/ and topo no go maps on separate topics, but I don't want to make a bigger mess right now as that would imply changing the localisation and emergency scripts too
